docs/samples/Microsoft.ML.Samples/Dynamic/Calibrator.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.ML.Calibrator;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public class CalibratorExample
+    {
+        public static void Calibration()
+        {
+            // Downloading the dataset from github.com/dotnet/machinelearning.
+            // This will create a sentiment.tsv file in the filesystem.
+            // You can open this file, if you want to see the data. 
+            string dataFile = SamplesUtils.DatasetUtils.DownloadSentimentDataset();
+
+            // A preview of the data. 
+            // Sentiment	SentimentText
+            //      0	    " :Erm, thank you. "
+            //      1	    ==You're cool==
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Read the data as an IDataView.
+            var reader = mlContext.Data.TextReader(new TextLoader.Arguments()
+            {
+                Separator = "tab",
+                HasHeader = true,
+                Column = new[]
+                    {
+                        new TextLoader.Column("Sentiment", DataKind.BL, 0),
+                        new TextLoader.Column("SentimentText", DataKind.Text, 1)
+                    }
+            });
+
+            // Read the data
+            var data = reader.Read(dataFile);
+
+            // Split the dataset into two parts: one used for training, the other to train the calibrator
+            var (trainData, calibrateData) = mlContext.BinaryClassification.TrainTestSplit(data, testFraction: 0.1);
+
+            // Featurize the text column through the FeaturizeText API. 
+            // Then append a binary classifier, setting the "Label" column as the label of the dataset, and 
+            // the "Features" column produced by FeaturizeText as the features column. 
+            var pipeline = mlContext.Transforms.Text.FeaturizeText("SentimentText", "Features")
+                    .Append(mlContext.BinaryClassification.Trainers.StochasticDualCoordinateAscent(
+                        labelColumn: "Sentiment", 
+                        featureColumn: "Features", 
+                        l2Const: 0.001f, 
+                        loss: new HingeLoss()));
```

Comment:
>loss: new HingeLoss() [](start = 24, length = 21)

`new TrainerInfo(calibration: !(_loss is LogLoss))`
You pass HingeLos,, so calibration in trainer should be true, and we should create calibrated predictor, but we got from transform only score column.
Why we don't have probability column by default?
Am I missing something?

Reply:
There is a reversed logic in the training, wrt inserting the Probability column. 

---
In reply to: [239303436](https://github.com/dotnet/machinelearning/pull/1816#discussion_r239303436) [](ancestors = 239303436)